draft-ietf-quic-multipath.md

```diff
@@ -381,6 +381,12 @@ that address. After receiving packets from the
 client on a new path, if the server decides to use the new path,
 the server MUST perform path validation ({{Section 8.2 of QUIC-TRANSPORT}})
 unless it has previously validated that address.
+A server can also decide not to perform path validation if it wants to deny the new path,
+or not perform it immediately if, e.g., no connection ID is yet available.
+If an endpoint decides to deny the establishment of a new path,
+this should be considered as a path closure and therefore requires sending a
+PATH_ABANDON frame on another path for the associated Path ID used
+by the client as specified in {{path-close}}.
```

Comment:
```suggestion
If an endpoint decides to deny the establishment of a new path,
it may not response to the path challenge. However,
this should be considered as a path closure and therefore requires sending a
PATH_ABANDON frame on another path for the associated Path ID used
by the client as specified in {{path-close}}.
```

Reply:
Okay, what I'm proposing here now is to remove the first sentence (see edit further above) and extend the second sentence to also say that you don't have to reply to the path challenge if you want to close. Does that work @huitema ?

What to do if you don't have a CID for this path is discussed further below. However, the text says this 

> If the server receives a PATH_CHALLENGE before receiving
> a MP_NEW_CONNECTION_ID for the specific path, it SHOULD
> ignore the PATH_CHALLENGE. Note that the
> MP_NEW_CONNECTION_ID might be sent in the same
> packet and in this case the PATH_CHALLENGE SHOULD
> be processed.

This is based on the discussion we have in issue #340 and added by PR #341. However, this seems wrong now as it should say that you can either close immediately by sending an abandon frame or wait for a bit. However, it should not be just ignored! Should we try to fix this part in this PR as well? I think I would prefer to leave it as it, merge this PR and create another issue for a new PR. Okay?

Reply:
You want to add something about not sending a PATH_RESPONSE frame. That's fine, but maybe we need first to agree on what we want to say. There are two parts:

* sending a PATH_RESPONSE or not. 
* sending PATH_ABANDON.

We had a discussion in the WG, and there is some plausible variation on either statement.

First, I think that the way to not accept a path is indeed to not send a path response. We agree on that, but RFC 9000 section 8.2.2 says: _On receiving a PATH_CHALLENGE frame, an endpoint MUST respond by echoing the data contained in the PATH_CHALLENGE frame in a PATH_RESPONSE frame. An endpoint MUST NOT delay transmission of a packet containing a PATH_RESPONSE frame unless constrained by congestion control._ That means we make a reference to the 8.2.2, and specify that if the receiver does not want to accept the path, it must not send the PATH_RESPONSE. (Of course, an endpoint could also send a PATH_RESPONSE and then send a PATH_ABANDON later -- there is not rule against that.)

Second, we had a discussion of whether the PATH_ABANDON shall be sent immediately. I think it depends on the reason why the endpoint cannot accept the new path. It may be a definitive reason, but it may also be a temporary reason, such as "I have not received a PATH NEW CID for that PATH yet." In the temporary case, it makes sense to not send the PATH_ABANDON immediately, just wait until the client repeats the PATH_CHALLENGE and acknowledge it then.

If for whatever reason the denying endpoint does not send the PATH_ABANDON immediately, the peer will trying repeating the challenge, and eventually will give up and PATH_ABANDON. At that point, the endpoint MUST send its own PATH_ABANDON.

Our own challenge is to convey all that in a couple of short paragraphs.